### ISSUE TITLE
usability: callstack UI should not scroll

### ISSUE THREAD

**Issue by Spongman** [microsoft/vscode#205142]
when debugging code with long callstack lines (eg c++ templates where line length can easily exceed 10,000 characters), much of the callstack UI is invisible without scrolling to the right making it next to useless:

![image](https://github.com/microsoft/vscode/assets/1088194/c1372e24-db25-4486-b1b1-694eb44f7ec5)

![image](https://github.com/microsoft/vscode/assets/1088194/97e6347e-01dc-4268-a913-c7073c6cedc1)

NOTE:
- the horizontal scrollbar is tiny.
- the filenames & line numbers (eg 'epoll_reator.ipp' 806:1) are not visible unless the window is scrolled all the way to the right
- the module/function names are not visible when the window is scrolled
- the 'Load More Stack Frames' UI is inaccessible, requiring precise scrolling to the middle of the window.

REQUEST:
- lock the filename/line number UI to the right-side of the VIEW (not the content)
- lock the "Load More Stack Frames" UI to the center of the VIEW (not the content)

**Comment by roblourens** [comment]
Agreed that it's not a good experience, but you would only see this with horizontal scrolling enabled, which isn't the default. I think that lots of our lists are not really optimized for horizontal scrolling.

**Comment by Spongman** [comment]
well, with `"workbench.list.horizontalScrolling": false,` it's _even_ less usable: 

![image](https://github.com/microsoft/vscode/assets/1088194/9e8b195f-a794-421e-ba70-fa172235dfc3)

- the right-side UI is _only_ visible when the line is short
- if the line is long, it's _impossible_ to see the filename/line number 
- if the line is long, it's _impossible_ to see the rest of the line

so, basically that setting switches between bad and worse.

**Comment by Spongman** [comment]
IMO, the way it _should_ works is this:
- the view should be horizonally scrollable
- the right-side UI should be visible for every line, should be right-justified to the VIEW (not affected by horizontal scroll), and should _overlap_ the list content (that _is_ scrolled).
- the "Load More Stack Frames" UI should be centered in the VIEW (not affected by horizontal scroll)

**Comment by roblourens** [comment]
I see. Those frame line names though 😳
